src/visualize/VoronoiGeometries.cpp

```diff
@@ -266,6 +266,10 @@ NuTo::Visualize::VoronoiGeometry TransformedVoronoi(const NuTo::IntegrationTypeB
 
 VoronoiGeometry NuTo::Visualize::VoronoiGeometryTriangle(const NuTo::IntegrationTypeBase& integrationType)
 {
+    for (int i = 0; i < integrationType.GetNumIntegrationPoints(); ++i)
+        if (integrationType.GetLocalIntegrationPointCoordinates(i).sum() > 1.)
+            throw Exception(__PRETTY_FUNCTION__, "Only 2D triangle types are supported");
+
```

Comment:
Ok, seems here would this shape thing be helpful, since tetrahedra would not throw and Quad, Line, Hexes with only one integration point at the center also not.